File: go/number-guessing-game/internal/game/game.go
package game

import "fmt"

func Game(secretNumber, attempts int) {
	remainder := attempts
	for attempts > 0 {
		var guess int
		fmt.Printf("%d", secretNumber)
		fmt.Printf("Enter your guess (%d attempts left): ", attempts)

		_, err := fmt.Scanf("%d", &guess)
		if err != nil {
			fmt.Println("Invalid input. Please enter a number.")
			fmt.Scanln()
			continue
		}

		if guess == secretNumber {
			fmt.Printf("Congratulations! You guessed the correct number in %d attempts.\n", remainder-attempts+1)
			break
		} else if guess < secretNumber {
			fmt.Printf("Incorrect! The number is greater than %d\n", guess)
		} else {
			fmt.Printf("Incorrect! The number is lesser than %d\n", guess)
		}

		attempts--
	}

	if attempts == 0 {
		fmt.Printf("Game over! The correct number was %d.\n", secretNumber)
	}
}

File: python/number-guessing-game/game.py
def play_game(secret_number, attempts):
    for attempt in range(1, attempts + 1):
        try:
            guess = int(input(f"Enter your guess ({attempts - attempt + 1} attempts left): "))
        except ValueError:
            print("Invalid input. Please enter a number.")
            continue

        if guess == secret_number:
            print(f"🎉 Congratulations! You guessed the correct number in {attempt} attempts.")
            return
        elif guess < secret_number:
            print(f"Incorrect! The number is greater than {guess}.")
        else:
            print(f"Incorrect! The number is lesser than {guess}.")

    print(f"💀 Game over! The correct number was {secret_number}.")
